lua/controller.lua

```diff
@@ -253,49 +258,68 @@ local function controller_get_drawer_index(pos, itemstring)
 	return drawer_net_index
 end
 
+local function controller_insert_to_empty_drawer(pos, stack)
+	local name = stack:get_name()
+	local drawer_net_index = controller_get_drawer_index(pos, name)
+
+	if not drawer_net_index["empty"] then return stack end
+
+	local drawer_pos = drawer_net_index["empty"]["drawer_pos"]
+	local visualid = drawer_net_index["empty"]["visualid"]
+	local content = drawers.drawer_get_content(drawer_pos, visualid)
+
+	-- If the drawer is still empty and the drawer entity is loaded, we will
+	-- put the items in the drawer
+	if content.name == "" and drawers.drawer_visuals[core.serialize(drawer_pos)] then
+		local leftover = drawers.drawer_insert_object(drawer_pos, stack, visualid)
+
+		-- Add the item to the drawers table index and set the empty one to nil
+		drawer_net_index["empty"] = nil
```

Comment:
`"empty"` is a valid item stack name. Potential conflict. How about a numeric index (perhaps `0`) instead, if you don't want to count the keys of this table.

Reply:
what about a pseudo item: "drawers:empty" would that work or should we keep short to minimize space used in meta?
I'm not a fan of mixing numerical and string indexes in tables.

Reply:
Valid node names are `[%w_]*`, that's equivalent to `[A-z0-9_]`. If you really want to mix two use-cases into a single table, and want to keep the string type as well, you could use an invalid character `;`. But not even this is a safe solution because `minetest.register_node(":NodeOverride-_operator;;,", {...})` is currently accepted as a node name.